google/cloud/internal/grpc_opentelemetry.cc

```diff
@@ -74,19 +74,33 @@ class GrpcClientCarrier
 };
 
 // We translate some keys, and modify binary values to be printable.
-std::pair<std::string, std::string> MakeAttribute(
+std::map<std::string, std::string> MakeAttributes(
     std::pair<std::string, std::string> kv) {
+  std::map<std::string, std::string> result;
   if (kv.first == ":grpc-context-peer") {
     // TODO(#10489): extract IP version, IP address, port from peer URI.
     // https://github.com/grpc/grpc/blob/master/src/core/lib/address_utils/parse_address.h
-    return {"grpc.peer", std::move(kv.second)};
+    // This is a hack-y solution until gRPC provides a way to parse the address:
+    // https://github.com/grpc/grpc/issues/35885
+    // The address should be in the format: host [ ":" port ]
```

Comment:
I don't think this isn't true. The peer string can also look like `ipv4:A.B.C.D:P` or `ipv6:[A:B:C:D:E:F]:P`

Reply:
BTW I asked grpc if they would expose the logic, but they said they could not bc it could be a security risk https://github.com/grpc/grpc/issues/35885

I was basing this off the URI spec which says a URI should be in the form user@host:port

Reply:
Do you have any reference for what a peer string looks like?

Reply:
I didn't find any.

We can sort of infer the possible peer strings from the gRPC unit tests, but eh.

https://github.com/grpc/grpc/blob/527557d03daf64c7bca5987dd167a0b02a3d2a01/test/core/address_utils/parse_address_test.cc#L168-L173

Reply:
Discussed with Brad. Maybe I can use ` man inet_pton` to parse the ip addresses